qiskit/transpiler/runningpassmanager.py

```diff
@@ -147,72 +112,38 @@ def run(
         """
-        return super().run(
-            in_program=circuit,
-            callback=_rename_callback_args(callback),
-            output_name=output_name or circuit.name,
-        )
-
-    def _run_base_pass(
-        self,
-        pass_: BasePass,
-        passmanager_ir: DAGCircuit,
-    ) -> DAGCircuit:
-        """Do either a pass and its "requires" or FlowController.
-
-        Args:
-            pass_: A base pass to run.
-            passmanager_ir: Pass manager IR, i.e. DAGCircuit for this class.
-
-        Returns:
-            The transformed dag in case of a transformation pass.
-            The same input dag in case of an analysis pass.
-
-        Raises:
-            TranspilerError: When transform pass returns non DAGCircuit.
-            TranspilerError: When pass is neither transform pass nor analysis pass.
-        """
-        pass_.property_set = self.property_set
-
-        if pass_.is_transformation_pass:
-            # Measure time if we have a callback or logging set
-            new_dag = pass_.run(passmanager_ir)
-            if isinstance(new_dag, DAGCircuit):
-                new_dag.calibrations = passmanager_ir.calibrations
-            else:
-                raise TranspilerError(
-                    "Transformation passes should return a transformed dag."
-                    "The pass %s is returning a %s" % (type(pass_).__name__, type(new_dag))
-                )
-            passmanager_ir = new_dag
-        elif pass_.is_analysis_pass:
-            # Measure time if we have a callback or logging set
-            pass_.run(FencedDAGCircuit(passmanager_ir))
-        else:
-            raise TranspilerError("I dont know how to handle this type of pass")
-        return passmanager_ir
+        property_set = PropertySet()
 
-    def _update_valid_passes(self, pass_):
-        super()._update_valid_passes(pass_)
-        if not pass_.is_analysis_pass:  # Analysis passes preserve all
-            self.valid_passes.intersection_update(set(pass_.preserves))
+        passmanager_ir = circuit_to_dag(circuit)
+        passmanager_ir = super().execute(
+            passmanager_ir=passmanager_ir,
+            property_set=property_set,
+            callback=callback,
+        )
 
+        out_circuit = dag_to_circuit(passmanager_ir, copy_operations=False)
+        out_circuit.name = output_name
 
-def _rename_callback_args(callback):
-    """A helper function to run callback with conventional argument names."""
-    if callback is None:
-        return callback
+        if property_set["layout"] is not None:
+            circuit._layout = TranspileLayout(
+                initial_layout=property_set["layout"],
+                input_qubit_mapping=property_set["original_qubit_indices"],
+                final_layout=property_set["final_layout"],
+            )
+        circuit._clbit_write_latency = property_set["clbit_write_latency"]
+        circuit._conditional_latency = property_set["conditional_latency"]
```

Comment:
The whole content of `RunningPassManager.run` is duplicated from `PassManager.execute`.  
Just to check: that's only for backwards compatibility, right?

Reply:
Yes, the method is also decorated with the `deprecate_func` decorator. Once this run method is removed we can drop this class.